<launch>
  <arg name="rvizconfig" default="$(find robotics-lab4)/rviz/urdf.rviz" />
  <arg name="perspective" default="$(find rqt_pincher_gui)/rqt_config/debug.perspective" />

  <node name = "pincher_node" pkg="robotics_lab5" type="pincher.py" output="screen"/>
  
  <node name="pincher_gui" pkg="rqt_gui" type="rqt_gui" respawn="false" output="screen" args="--perspective-file $(arg perspective) "/>

  <arg name="dinamixel_is_connected"                default="false"/>

<node name = "dinamixel_dummy" pkg="robotics-lab4" type="dinamixelDummy.py" 
      output="screen" if="$(eval arg('dinamixel_is_connected')==false)"/>

 <group if="$(eval arg('dinamixel_is_connected'))">

  
<!-- Port -->
  <arg name="usb_port"                default="/dev/ttyUSB0"/>
  <!-- Dynamixel FTDI baudrate -->
  <arg name="dxl_baud_rate"           default="1000000"/>
  <!-- Nodes Namespace -->
  <arg name="namespace"               default="dynamixel_workbench"/>
  <!-- Mics options -->
  <arg name="use_moveit"              default="false"/>
  <arg name="use_joint_state"         default="true"/>
  <arg name="use_cmd_vel"             default="false"/>

  <!-- Load the motors configuration -->
  <param name="dynamixel_info"          value="$(find robotics-lab4)/config/basic.yaml"/>
  
  <node name="$(arg namespace)" pkg="dynamixel_workbench_controllers" type="dynamixel_workbench_controllers"
        required="true" output="screen" args="$(arg usb_port) $(arg dxl_baud_rate)">
    <param name="use_moveit"              value="$(arg use_moveit)"/>
    <param name="use_joint_states_topic"  value="$(arg use_joint_state)"/>
    <param name="use_cmd_vel_topic"       value="$(arg use_cmd_vel)"/>
    <rosparam>
      publish_period: 0.010
      dxl_read_period: 0.010
      dxl_write_period: 0.010
      mobile_robot_config:                <!--this values will be set when 'use_cmd_vel' is true-->
        seperation_between_wheels: 0.160  <!--default value is set by reference of TB3-->
        radius_of_wheel: 0.033            <!--default value is set by reference of TB3-->
    </rosparam>
    <remap from="/dynamixel_workbench/joint_trajectory" to="joint_trajectory"/>
  </node>
</group>
 
</launch>
